content/ui-patterns/messaging.mdx

```diff
@@ -7,25 +7,96 @@ import {Box} from '@primer/react'
 
-**Feedback**: communicates something to the user based on an action they've taken or a change in context
+**System updates**: These types of message originate from GitHub and are not user initiated. A global site wide announcement (outage, missing payment details, critical and blocking) or messaging related to user account/permissions should use the **Announcement** component. System announcements related to a specific product area such as a single sign-on message should use the **Banner** component. System update type messages cannot be dismissed until the issue is resolved, or may be perminent if they are purely informative and not critical.
 
-- _Toasts or flash alerts are best suited for this type of messaging._
+(image here)
 
-**Highlights**: bring attention to how and why to engage with a feature
+**Feedback**: Communicates the result of a user action (subitting a form, toggling a setting, etc.) Feedback can be positive, negative, or a warning. Feedback is shown in either a **Banner** or **Inline message**. For forms, an inline message is used for individual field validation which may be combined with a banner if multiple fields have errors. Banners are typically placed near the top of the page, but still within the body content. Inline messages are placed near the action they are related to.
 
-- _Popovers are best suited for this type of messaging._
+(image here)
 
-**Updates**: inform the user of necessary updates or changes to the product as it relates to them.
+**Awareness**: Communicates information that is relevant to the user but not necessarily related to an action they've taken. This type of message is typically used to provide context or additional information about a feature or product. A **Callout** might be used to suggest an action or helpful tip, while a **Popover** is typically used to announce a new feature and offers a link to learn more.
 
-- _Flash alerts are best suited for this type of messaging._
+(image here)
+
+## Message states
+
+### Info
+The info state should be used to provide additional information that when missed to not prevent users from performing their desired actions.
+An info may be:
+- a setting that impacts the current view
+- a feature that is now available for you
+- additional details about a section or action that are more important than normal
+
+### Warning
+The warning state should be used if a user action is required prevent a problem in the future. Additionally if information is provided that is very important so that missing this information may lead to unwanted consequences.
+A warning can be:
+- 2FA will be required soon and the users is not using it
+- performing this action will have drastic effects e.g. deleting or archiving a repo, etc.
+- performing an action will incur costs for the first time
+- the system is experiencing temporary problems that impact the quality of service but do not make working completly impossible
+
+
+### Error
```

Comment:
I've seen "danger"/red variant used to communicate when an action is super destructive or something needs immediate attention. Is that appropriate, and would that fall under this state?

Reply:
Good question.

One the one side, _technically_ it would be invalid because it is not an error state it is a warning state.
On the other side it is a pattern our users have grown accustomed too, so it may be dangerous to change this.

@langermank what do you think?

I feel like it would make sense to extend this definition to include _destructive actions_, this is also why we use the danger button, which is red as well.